apps/notifications/signals.py

```diff
@@ -1,22 +1,57 @@
-from django.db.models.signals import post_save
+from django.db.models import signals
 from django.dispatch import receiver
 
 from adhocracy4.actions.models import Action
 from adhocracy4.actions.verbs import Verbs
 from adhocracy4.comments.models import Comment
 
-from apps.ideas.models import IdeaSketch, Proposal
+from apps.ideas.models import Idea, IdeaSketch, Proposal
+from apps.journeys.models import JourneyEntry
 
 from . import emails
 
 
-@receiver(post_save, sender=Action)
+@receiver(signals.post_save, sender=Action)
 def send_notification(sender, instance, created, **kwargs):
     action = instance
     if action.verb == Verbs.ADD.value:
+        modelcls = action.obj_content_type.model_class()
 
-        if (action.obj_content_type.model_class() is IdeaSketch
-                or action.obj_content_type.model_class() is Proposal):
+        if (modelcls is IdeaSketch):
             emails.SubmitNotification.send(action.obj)
-        elif action.obj_content_type.model_class() is Comment:
+        elif (modelcls is Proposal):
+            emails.SubmitNotification.send(action.obj)
+            emails.NotifyFollowersOnNewProposal.send(action)
+        elif (modelcls is Comment):
             emails.NotifyCreatorEmail.send(action)
+            emails.NotifyFollowersOnNewComment.send(action)
+        elif (modelcls is JourneyEntry):
+            emails.SubmitJourneyNotification.send(action.obj)
+            emails.NotifyFollowersOnNewJourney.send(action)
+
+
+@receiver(signals.pre_save, sender=Idea)
+def identify_status_changes(sender, instance, **kwargs):
+    old_idea = Idea.objects.filter(pk=instance.pk).first()
```

Comment:
Nicely done for updates, but how does this work when I create a new proposals? In that case `instance.pk` and `old_idea` should be None and this should be crashing.

Reply:
I also remember that there was some issues with `save` not being by changes on the admin action (due to using `qs.update`). Doesn't this approach still have the same issue?

Reply:
> Nicely done for updates, but how does this work when I create a new proposals? In that case instance.pk and old_idea should be None and this should be crashing.

AFAIK, changes in the proposal model do not trigger that signal. So, this might not be an issue. Or am I missing something?

> I also remember that there was some issues with save not being by changes on the admin action (due to using qs.update). Doesn't this approach still have the same issue?

Good Point, this only occurs if the admin uses the the Action-Dropdown-Voodoo in the idea list view in the admin interface, not if he sets the check box in the idea detail view. However, still I missed this issue in total and I'll love to provide a fix for that.

Reply:
Commit https://github.com/liqd/a4-advocate-europe/pull/468/commits/10dbeaa25e5c1f2161477b548f27504b83c70d2f tries to fix this.